### ISSUE TITLE
The issue of Step1

### ISSUE THREAD

**Issue by pxxiao-hz** [sakshar/rambler#1]
Hello. 
I tried rambler today and ran into some trouble.

I don’t know the meaning of step1, what do `scripts/mapReads.sh`, reads and genome mean?

I tried all HiFi reads and the genome from hifiasm, the memory occupied about 300G, and then the script was killed by the system.
The problem occurs when using NucPlot.py to plot. `python3 $NUCPLOT/NucPlot.py -w 32 $H.sorted.nosecondary.bam $H.png`.

I used the whole genome, could you explain `Determine the reads corresponding to repetitive regions`? 
Thanks.

**Comment by sakshar** [comment]
Hi,

Thank you for reaching out. The problem you are facing that may arise when the genome contains many contigs or chromosomes. The problem occurs because `NucPlot.py` tries to put the plots for all the contigs/chromosomes into a single file and it then crashes. One way to  to get to the sequence coverage plot avoiding this issue is producing the plots in separate files for each of the contigs/chromosomes. I am attaching a script that can produce plots for each contig/chromosome separately. Please use that for generating the plots. If it still doesn't solve your issue, don't hesitate to get back to me.

`Determine the reads corresponding to repetitive regions`, this refers to the regions in the plots where there will be unusual spikes. You first inspect the plots to see if you find any spikes there. If you find one, note down the contig/chromosome id it belongs to and also the indices/positions between which it appears in the sequence. You then follow the `mapReads.sh` script to extract the reads mapping to those particular regions. Hope it answers your query.
[mapReads_separate_files.sh.zip](https://github.com/sakshar/rambler/files/12653727/mapReads_separate_files.sh.zip)

**Comment by pxxiao-hz** [comment]
I have got the coverage plot. The other sequences are still running.

<img width="1787" alt="image" src="https://github.com/sakshar/rambler/assets/76642822/17cf6989-3030-4daf-a77b-6398ec781a50">

If there are any questions later, I will provide feedback in time.

Thanks.

**Comment by pxxiao-hz** [comment]
`only after calculating the mean and sd from the excel file`, Does this file need to be filtered? High frequency kmer will have a big impact.
[kmer.xlsx](https://github.com/sakshar/rambler/files/12679210/kmer.xlsx)

**Comment by sakshar** [comment]
Yes, you need to filter out the kmers which are low frequency and also the kmers which are high frequency. It is better to first visualize the distribution. Then you will see a nice normal distribution curve as shown in the attached file. I have taken only the window around the distribution curve for calculating the mean and the standard deviation, specifically from 21 to 120. Let me know if you have any more questions.
[test_kmer.xlsx](https://github.com/sakshar/rambler/files/12681091/test_kmer.xlsx)

**Comment by pxxiao-hz** [comment]
When I run `run.sh`, `python3 ./from_read_unikmer_to_clusters.py $READ_FILE $KMER_FILE $OUT_DIR $REF_SIZE $k_len $tolerance $threshold` was killed. The memory uses 500G, which reached our server limited. 
This is my command: 
`bash run.sh -r ../data_potato/hifi-reads.fasta -u ../step2/test.unikmers -o out_step3 -s 700000000`

I would like to know Rambler's memory usage on the test set.

**Comment by sakshar** [comment]
On the test set RAmbler's memory usage was 633 MB. However, RAmbler is not designed for performing whole genome scale assembly. You are expected to run assembly in smaller subsets of HiFi reads including only the reads mapping to the unresolved repeat regions existing in the draft whole genome sequence. From your command, I suspect that you are trying to re-assemble potato genome and you are passing the complete set of reads altogether. I am afraid that RAmbler will fail, if you attempt at executing it in this fashion. Please try to find the subset of reads that maps to unresolved repeats in the existing genome and then re-assemble only those segments separately and then stitch the newly obtained subsequences back to the original draft genome.

From our tests, we can confirm that RAmbler can handle about 10,000 HiFi reads in a single execution. If you go beyond that the memory resources could be exhausted. Hope this helps and clarifies your concern.
